core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdbutil.go

```diff
@@ -98,18 +98,16 @@ func createCouchInstance(config *ledger.CouchDBConfig, metricsProvider metrics.P
 	return couchInstance, nil
 }
 
-//checkCouchDBVersion verifies CouchDB is at least 2.0.0
 func checkCouchDBVersion(version string) error {
-
-	//split the version into parts
-	majorVersion := strings.Split(version, ".")
-
-	//check to see that the major version number is at least 2
-	majorVersionInt, _ := strconv.Atoi(majorVersion[0])
-	if majorVersionInt < 2 {
-		return errors.Errorf("CouchDB must be at least version 2.0.0. Detected version %s", version)
+	couchVersion := strings.Split(version, ".")
+	majorVersion, _ := strconv.Atoi(couchVersion[0])
+	minorVersion, _ := strconv.Atoi(couchVersion[1])
+	if majorVersion < 2 {
+		return errors.Errorf("CouchDB v%s detected. CouchDB must be at least version 2.0.0", version)
+	}
+	if majorVersion != 3 && minorVersion < 1 {
+		logger.Warnf("CouchDB v%s detected. CouchDB versions before 3.1.0 are unsupported.", version)
```

Comment:
Is this ok to just log a warning instead of returning an error?

Reply:
It is, this was ultimately what we decided. Technically 2.x Couch is highly tested in previous releases, and customers can continue to use it, it is just not our officially supported version anymore.

Reply:
This will come with a strong statement of recommendation for users to migrate to Couch 3 in the release notes

Reply:
On that note this check won't actually log a warning for 3.0.x or 2.1.x or for that matter any 2.x version where x >= 1.